From 4090e233ae610b3c07960a248c899ae03309f52e Mon Sep 17 00:00:00 2001
From: Jesse Chan <jc@lineageos.org>
Date: Thu, 18 Oct 2018 18:21:35 -0700
Subject: [PATCH 15/28] HWComposer: HWC2: allow SkipValidate to be force
 disabled

HWC on some devices does not support SkipValidate.
On earlier versions, hasCapability(HWC2::Capability::SkipValidate)
will be called to check if SkipValidate is available.

This check has been removed in Pie and now surfaceflinger
rely on HWC to fallback to validate when there is any client layer.
But HWC is not reliable as some vendors do not implement fallback at all.

Thus, this change adds a flag to allow SkipValidate to be force disabled.

Change-Id: I61f69bd82cad74820e42246182c3d5b131137831

# Conflicts:
#	build/soong/android/variable.go
---
 build/soong/android/variable.go | 4 ++++
 build/soong/soong_config.mk     | 1 +
 2 files changed, 5 insertions(+)

diff --git a/build/soong/android/variable.go b/build/soong/android/variable.go
index 029a518f..5ababc6d 100644
--- a/build/soong/android/variable.go
+++ b/build/soong/android/variable.go
@@ -25,6 +25,9 @@ type Product_variables struct {
 	}
 	Supports_legacy_hw_fde struct {
 		Cflags []string
+        }
+	Hwc2_no_skipvalidate struct {
+		Cppflags []string
 	}
 	Target_process_sdk_version_override struct {
 		Cppflags []string
@@ -52,6 +55,7 @@ type ProductVariables struct {
 	Apply_msm8974_1440p_egl_workaround  *bool `json:",omitempty"`
 	Bootloader_message_offset  *int `json:",omitempty"`
 	Has_legacy_camera_hal1  *bool `json:",omitempty"`
+	Hwc2_no_skipvalidate  *bool `json:",omitempty"`
 	Java_Source_Overlays *string `json:",omitempty"`
 	Should_skip_waiting_for_qsee  *bool `json:",omitempty"`
 	Specific_camera_parameter_library  *string `json:",omitempty"`
diff --git a/build/soong/soong_config.mk b/build/soong/soong_config.mk
index 1444d0bc..a0535e6d 100644
--- a/build/soong/soong_config.mk
+++ b/build/soong/soong_config.mk
@@ -8,6 +8,7 @@ $(call add_json_str_omitempty, Additional_gralloc_10_usage_bits, $(TARGET_ADDITI
 $(call add_json_bool, Apply_msm8974_1440p_egl_workaround, $(filter true,$(TARGET_MSM8974_1440P_EGL_WORKAROUND)))
 $(call add_json_val_default, Bootloader_message_offset, $(BOOTLOADER_MESSAGE_OFFSET), 0)
 $(call add_json_bool, Has_legacy_camera_hal1, $(filter true,$(TARGET_HAS_LEGACY_CAMERA_HAL1)))
+$(call add_json_bool, Hwc2_no_skipvalidate, $(filter true,$(TARGET_HWC2_NO_SKIPVALIDATE)))
 $(call add_json_str, Java_Source_Overlays, $(JAVA_SOURCE_OVERLAYS))
 $(call add_json_bool, Should_skip_waiting_for_qsee, $(filter true,$(TARGET_KEYMASTER_SKIP_WAITING_FOR_QSEE)))
 $(call add_json_str, Specific_camera_parameter_library, $(TARGET_SPECIFIC_CAMERA_PARAMETER_LIBRARY))
-- 
2.17.1

